crates/ty_python_semantic/src/types.rs

```diff
@@ -2746,6 +2746,21 @@ impl<'db> Type<'db> {
             instance.display(db),
             owner.display(db)
         );
+        match self {
+            Type::Callable(callable) if callable.is_function_like(db) => {
+                // For "function-like" callables, model the the behavior of `FunctionType.__get__`:
```

Comment:
It is a shortcut to model this in `try_call_dunder_get`. If we want to be really precise, we should instead create a new method-wrapper type variant for the synthesized `__get__` method of these callables. This would be returned from `find_name_in_mro`. This would allow us to correctly model the behavior of explicit `SomeDataclass.__init__.__get__` calls. The occurrence of this seems rather unlikely, so I did not introduce a new type variant just for this. But if we would rather model this completely correctly, I'm happy to make the necessary changes.

Reply:
Could you add that brief summary as a comment, so we know what to do if it does come up in the future?